var/spack/repos/builtin/packages/cgns/package.py

```diff
@@ -42,6 +42,8 @@ class Cgns(CMakePackage):
     variant('legacy',     default=False, description='Enable legacy options')
     variant('mem_debug',  default=False, description='Enable memory debugging option')
 
+    depends_on('cmake@3.8:', when='@4.2:', type=('build'))
+    depends_on('cmake@2.8:', when='@:4.1.99', type=('build'))
```

Comment:
I think you can avoid parenthesis:
```suggestion
    depends_on('cmake@3.8:', when='@4.2:', type='build')
    depends_on('cmake@2.8:', when='@:4.1.99', type='build')
```

Reply:
Right, there is no need for a tuple.